packages/leancode_contracts/lib/src/types/binary.dart

```diff
@@ -0,0 +1,40 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+/// Stores binary data as a base64 encoded string.
+class Binary {
+  /// Constructs [Binary] given a list of bytes.
+  Binary(List<int> bytes) : this.fromJson(base64Encode(bytes));
```

Comment:
Keeping the value inside `Binary` as encoded makes the creation of `Binary` quite heavy. More than unnamed ctor should be. It also makes the usability of `Binary` as transit-focused. I'm not fully aware how the Flutter ecosystem approaches this, but `List<int>` seems to be a more _usable_ underlying type.

Reply:
> It also makes the usability of Binary as transit-focused

That was the idea behind this decision. It would be quite wasteful to decode base64 from all responses.

> I'm not fully aware how the Flutter ecosystem approaches this

Usually it is `Uint8List` downcasted to `List<int>`

I can't think of a good way to keep a bytes array as the underlying type without performing wasteful decodes/encodes :/